eureka/src/main/java/com/linecorp/armeria/internal/common/eureka/InstanceInfo.java

```diff
@@ -91,6 +93,7 @@ public InstanceInfo(@Nullable @JsonProperty("instanceId") String instanceId,
                         @JsonProperty("status") InstanceStatus status,
                         @Nullable @JsonProperty("homePageUrl") String homePageUrl,
                         @Nullable @JsonProperty("statusPageUrl") String statusPageUrl,
+                        @Nullable @JsonProperty("healthCheckPath") String healthCheckPath,
```

Comment:
How about adding another package-private constructor that is not a JSON creator instead of adding the property to this constructor? `InstanceInfoBuilder` will call that package-private constructor with the configured `healthCheckPath `.

I agree that `healthCheckPath` is needed but I'm not sure if we can add this with the JSON property which is not compatible with the upstream: https://github.com/Netflix/eureka/blob/2b09d77534eae703a36f79f99b27b4d6ed66c3f8/eureka-client/src/main/java/com/netflix/appinfo/InstanceInfo.java#L165-L191

The best way to solve this is adding the `healthCheckPath` field to the upstream but it's probably not going to work because upstream isn't maintained well.
The upstream also uses the same approach which allows to set the path from the builder:
https://github.com/Netflix/eureka/blob/2b09d77534eae703a36f79f99b27b4d6ed66c3f8/eureka-client/src/main/java/com/netflix/appinfo/InstanceInfo.java#L663-L666

Reply:
thanks for the advice, I'll run from here @minwoox!